csrc/device_lower/validation.cpp

```diff
@@ -410,6 +411,362 @@ class ExprValidator : public OptOutDispatch {
           (*it)->toString());
     }
   }
+
+  // A merge is acceptable its inputs can be traced back to IDs in the logical
```

Comment:
This PR is already too large. Let's remove this validation from this PR and create a separate PR with specific tests.